xchange-coingi/src/main/java/org/knowm/xchange/coingi/dto/CoingiException.java

```diff
@@ -0,0 +1,47 @@
+package org.knowm.xchange.coingi.dto;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+import java.util.List;
+import java.util.Map;
+import si.mazi.rescu.HttpStatusExceptionSupport;
+
+public class CoingiException extends HttpStatusExceptionSupport {
+  private List<Map<String, Object>> errors;
+
+  public CoingiException(@JsonProperty("errors") Object errors) {
```

Comment:
I'm 90% sure that the client library cannot handle constructor injection for JSON propierties in the case of exceptions and you'll get errors = null here every time

adding an integration test like this one would check that: https://github.com/walec51/XChange/blob/d7f639836bf84546743486465152d52f26c3cdb4/xchange-bittrex/src/test/java/org/knowm/xchange/bittrex/service/marketdata/MarketDataIntegrationTest.java#L37

if I'm correct you shoud

- annotate the class attribute errors
- implement a bean style getter and setter for it
- override the getMessage and place your message building code there

Reply:
I've tested this (and verified once more in a debugger). No problem here. It's working nicely.

Reply:
then please add this:

```
  @Test
  public void invalidCurrencyPairForTradesFetchTest() throws Exception {
    Throwable excepton =
        catchThrowable(
            () -> marketDataService.getTrades(new CurrencyPair("NOT_EXISTING_CODE", "USD")));
    assertThat(excepton).isExactlyInstanceOf(CurrencyPairNotValidException.class);
  }
```

to you integrations tests to ease my paranoia :)